Mage.Sets/src/mage/cards/c/CleavingReaper.java

```diff
@@ -0,0 +1,84 @@
+package mage.cards.c;
+
+import java.util.List;
+import java.util.UUID;
+import mage.MageInt;
+import mage.abilities.Ability;
+import mage.abilities.common.ActivateIfConditionActivatedAbility;
+import mage.abilities.condition.Condition;
+import mage.abilities.costs.common.PayLifeCost;
+import mage.abilities.effects.common.ReturnSourceFromGraveyardToHandEffect;
+import mage.constants.SubType;
+import mage.abilities.keyword.FlyingAbility;
+import mage.abilities.keyword.TrampleAbility;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.CardType;
+import mage.constants.Zone;
+import mage.game.Game;
+import mage.game.permanent.Permanent;
+import mage.watchers.common.PermanentsEnteredBattlefieldWatcher;
+
+/**
+ *
+ * @author weirddan455
+ */
+public final class CleavingReaper extends CardImpl {
+
+    public CleavingReaper(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{3}{B}{B}");
+
+        this.subtype.add(SubType.ANGEL);
+        this.subtype.add(SubType.BERSERKER);
+        this.power = new MageInt(5);
+        this.toughness = new MageInt(3);
+
+        // Flying
+        this.addAbility(FlyingAbility.getInstance());
+
+        // Trample
+        this.addAbility(TrampleAbility.getInstance());
+
+        // Pay 3 life: Return Cleaving Reaper from your graveyard to your hand.
+        // Activate this ability only if you had an Angel or Berserker enter the battlefield under your control this turn.
+        Ability ability = new ActivateIfConditionActivatedAbility(
+                Zone.GRAVEYARD, new ReturnSourceFromGraveyardToHandEffect(), new PayLifeCost(3), CleavingReaperCondition.instance
+        );
+        this.addAbility(ability, new PermanentsEnteredBattlefieldWatcher());
```

Comment:
It will be useful to add `ConditionHint` about return condition ("Can return from graveyard"). So user able to see is it possible to return card before real card goes to graveyard.

Reply:
Done.